### PR TITLE
Adding Full CRUD operations on Ratings to Barkbaud

### PR DESCRIPTION
**Pull request by Blackbaud-CameronBeacham**

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -329,6 +329,298 @@
...
+    /**
+     *
+     * @name getDogRating
+     * @param {Object} request
+     * @param {String} request.param.dogId
+     * @param {String} request.param.dogRatingId
+     * @param {Object} response
+     */
+    function getDogRating(request, response) {
+        var lookup;
```
Comment (Blackbaud-SteveBrush) [inline server/routes/api.js]: Just being picky: `var lookup` could be placed inside the `.exec(` function below, to reflect its true scope.

File: server/routes/api.js [hunk 2]
```diff
@@ -329,6 +329,298 @@
...
+    }
+
+    /**
+     *
+     * @name getDogRatingCategories
+     * @param {Object} request
+     * @param {Object} response
+     */
+    function getDogRatingCategories(request, response) {
+        var dogCategories = ['House breaking', 'Activity level', 'Obedience', 'Motivated by', 'Estimated age', 'Favorite toy', 'Sheds', 'Allowance', 'Adoption date', 'Allowed treats per day']
```
Comment (Blackbaud-SteveBrush) [inline server/routes/api.js]: Missing a semicolon at the end of this array.

**Comment by Blackbaud-SteveBrush** [comment]
You'll notice that the server (this repo) contains a `ui` folder. This holds the most recent front-end build. In other words, `barkbaud-ui` repo is only consumed during development. Mind running `grunt build` locally and committing the result (after `blackbaud-ui` is approved).

File: server/routes/api.js (hunk 1 above)
Comment (Blackbaud-CameronBeacham) [reply server/routes/api.js]: Fixed

File: server/routes/api.js (hunk 2 above)
Comment (Blackbaud-CameronBeacham) [reply server/routes/api.js]: Fixed

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**